feat(Day 8): Class-based components

ruchamahabal opened the pull request:
https://endearing-pony-0aaded.netlify.app/

- About Us page Demonstrating class-based component lifecycle with Parent (AboutUs), Child (AboutUsChildSection), and NestedChild components
   - constructor
   - render
   - componentDidMount
   - componentDidUpdate
   - componentWillUnmount
- Using and updating states
- Passing props

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/24353136/208398847-e6579760-ceec-4486-828f-16718a31000c.png">

## componentWillUnmount demo using toggle

![Kapture 2022-12-19 at 15 35 47](https://user-images.githubusercontent.com/24353136/208400679-fbaf4b55-9943-4867-ae82-ec566298be4b.gif)

## Component lifecycle

<img width="1440" alt="Screen Shot 2022-12-19 at 3 29 34 PM" src="https://user-images.githubusercontent.com/24353136/208399927-3cfab998-e473-42b3-8020-0684ef0eef87.png">

ruchamahabal committed:
chore: Initialize app

ruchamahabal committed:
feat: Class-based component lifecycle

- demonstrated using Parent, Child, NestedChild components

ruchamahabal committed:
feat: use and update states and pass props in class components

ruchamahabal committed:
feat: demonstrate componentWillUnmount using toggle

ruchamahabal committed:
fix: props destructuring in child component

ruchamahabal committed:
refactor: separate out components

ruchamahabal committed:
feat: add nav links

ruchamahabal committed:
docs: class components are lifecycle methods

ruchamahabal commented:
Deployed on netlify, not on github pages: https://endearing-pony-0aaded.netlify.app/

ruchamahabal commented:
Note for the reviewer: Just check the `class_components` directory, readme and App.js. Other things are almost the same as before.

ap221882 commented on 8_class_components/src/App.js:
```diff
@@ -0,0 +1,60 @@
+import MemberList from "./components/MemberList.js";
+import MemberDetails from "./components/MemberDetails";
+import Error from "./components/Error.js";
+
```
If trying to group your imports, can add one line comment that by which criteria you're putting them separately, i.e., Local, Global, UI, styles, etc. These things will help when you've dozens of imported modules.

ap221882 commented on 8_class_components/src/App.js:
```diff
@@ -0,0 +1,60 @@
+						index: true,
+						element: (
+							<AboutUsChildSection
+								team_members="10"
```
Format the code here

ruchamahabal replied:
Prettier auto-formatted it 🙈. Any suggestion?

ap221882 replied:
No comments 🥹

ap221882 commented on 8_class_components/src/components/Card.js:
```diff
@@ -0,0 +1,58 @@
+			</div>
+			<div className="card-footer social-icon">
+				<a href={html_url}>
+					<svg
```
put svgs in separate file in assets folder. Keeps your code more clean.

ap221882 commented on 8_class_components/src/components/Error.js:
```diff
@@ -0,0 +1,13 @@
+	return (
+		<div className="centered-div">
+			<h2>Oops! Something went wrong!</h2>
+			<p>{`${errorData.statusText} : ${errorData.message}`}</p>
```
Put an || operator, either of the value can come during the error

ap221882 commented on 8_class_components/src/components/Error.js:
```diff
@@ -0,0 +1,13 @@
+import { useRouteError } from "react-router-dom";
+
+const Error = () => {
+	const errorData = useRouteError();
```
Have a blank line before return if your component involves some logic

ap221882 commented on 8_class_components/src/components/MemberDetails.js:
```diff
@@ -0,0 +1,92 @@
+
+	useEffect(() => {
+		getMemberDetails();
+	}, []);
```
add respective dependencies that are involved.

ruchamahabal replied:
No dependencies to be specified in this case, I guess. Correct me if I am wrong.

ap221882 commented on 8_class_components/src/components/SearchBar.js:
```diff
@@ -0,0 +1,54 @@
+ * returns filtered members that match name/designation with searchText
+ * @param {string} searchText text used to match for filtering
+ * @param {object} data data to filter from
+ */
```
can add @return too

ap221882 commented on 8_class_components/src/style.css:
```diff
@@ -0,0 +1,243 @@
+	font-size: 50px;
+}
+
+button {
```
Add this html tag selector to the top having global style and then add custom styles using classes according to different scenarios of the respective button.

ap221882 commented on 8_class_components/src/components/class_components/NestedChild.js:
```diff
@@ -0,0 +1,59 @@
+			<>
+				{course_name} course takes you from zero to hero in {duration}{" "}
+				months with hands-on coding and theory assignments,
+				<br /> interactive lectures, code reviews and a lot more!
```
can style through css, no need of br tag

ap221882 reviewed (commented):
Very beautifully written. Commenting is awesome. Kudos Rucha!!

Edit1: CSS is also good

ruchamahabal committed:
refactor: docs, styling, assets

- group imports with comments

- separate out svg in assets

- handle empty error data

- add return spec for function comments

- move button css to global css

- remove br tag and handle spacing using css

ruchamahabal committed:
refactor: fix delay in search - use onKeyUpevent

Co-authored-by: Utpal Sangwan <81686012+vickydarlinn@users.noreply.github.com>

ruchamahabal commented:
thank you for your valuable feedback @ap221882 🙏🏻

Pujarini commented:
Great looking UI ❤️ !

Pujarini commented on 8_class_components/src/components/Navbar.js:
```diff
@@ -1,11 +1,23 @@
+
+			<div className="nav-links">
+				<Link to="/" className="link">
+					Home
```
can you use prettier for formatting?

ruchamahabal replied:
Yes, prettier auto-formatted it this way

ruchamahabal commented:
> Great looking UI ❤️ !

Thanks for reviewing ❤️
